ENH: Creating an FSL-free version of MRIQC

NPann opened the pull request:
Replaces #807 , as I can't write that branch.

The plan is to minimize the dependency on FSL, and provide an alternative compute graph that is completely free of FSL.
From #807, fixed MD5 sum check, IQM check and set ANTs seed.

oesteban commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mriqc/workflows/anatomical.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/poldracklab/mriqc/pullRequest?prid=4677414&bid=15649255)

NPann commented:
Not sure why the circleCI checks do not show up here. But passing the CICD of my fork as can be shown [here](https://circleci.com/gh/NPann/mriqc/tree/enh%2Ffsl-free).

dorianps commented:
Adding comment to get notification when merged,

NPann commented:
@oesteban, happy new year! Any chance to get your inputs on this one?

NPann commented:
@oesteban, any chance to get your inputs one this PR? Or do you think we should just work from a fork?

oesteban commented:
Hi @NPann, I have blocked 2h of my calendar next Tue 24th to finish this up. Sorry, I'm snowed under quite a long backlog. Having you guys working off of a fork without a stronger reason other than my inability to go through your contribution seems unfortunate, I'll try to be more responsive (and bring in people from the community to help put the project forward).

Just so you know, MRIQC will soon be migrated under https://github.com/nipreps/ as part of the migration to a more community-driven approach (see www.nipreps.org).

oesteban commented:
BTW, I was checking the reports from https://circleci.com/workflow-run/a2f69a9d-cf02-4c1c-9a27-24deba6889a1 - please let me know if that is not correct.

NPann commented on mriqc/workflows/anatomical.py:
```diff
@@ -427,13 +445,13 @@ def headmsk_wf(name='HeadMaskWorkflow', use_bet=True):
     outputnode = pe.Node(niu.IdentityInterface(fields=['out_file']), name='outputnode')
 
     if use_bet or not has_dipy:
```
Solved by 5f98a18

NPann commented on mriqc/workflows/anatomical.py:
```diff
@@ -427,13 +445,13 @@ def headmsk_wf(name='HeadMaskWorkflow', use_bet=True):
     outputnode = pe.Node(niu.IdentityInterface(fields=['out_file']), name='outputnode')
 
```
Solved by 5f98a18

NPann commented:
@oesteban , Thanks for the comment. I indeed screwed up something with the parsing of the USE_FSL env var. I think this time I got it right this time. CI report at https://app.circleci.com/pipelines/github/NPann/mriqc/21/workflows/b79c9ffb-564d-45d4-b192-24085e909cba

The FSL license does not seem to allow for commercial redistribution. I created a fsl free dockerfile as well and added bold and T1w tests with it (which matches the docker image with fsl but USE_FSL=0). Updated circleCI to publish that image to dockerhub as well. If all look good, I can update the doc accordingly.

NPann commented:
Thanks @oesteban.
* I checked the segmentation and the main discrepancy seems to be for sub-50137. Any clue where to start to fix this?
* Re the classifier, I think this is something we are pretty well equipped to do at Flywheel. I'll request access to ABIDE, download ds030 ([this](https://openneuro.org/datasets/ds000030/versions/00016) version is the right one?) and dig into what we can do. Likely require to fix the segmentation issue aforementioned. 
* I'll split the fMRI into another PR.

effigies commented:
Was this superseded by #842?

NPann commented:
> Was this superseded by #842?

I don't think so. #842 was addressing the FMRI path. This one was re-focused on anatomical scans but still had a few issues as mentioned by Oscar. Unclear about path forward here.

effigies commented:
Ah, thanks for clarifying. I would say the next steps would probably be to merge current `master` (or rebase, if that's your preference), make sure tests are passing, and then re-ping.

But I'm not really deeply involved in MRIQC (just doing some maintenance), so @oesteban may have more specific thoughts.

oesteban commented:
It seems this is finally happening - only one remaining use of BET in the anatomical workflow (to generate the "outskin" mask).

After that, the only action remaining is replacing MELODIC -- maybe @eilidhmacnicol has the humor to take a stab at it and replace it with some nilearn ICA.

Closing this PR as it has become stale. Apologies for my inaction over time @NPann .
